string.d.ts

```diff
@@ -0,0 +1,27 @@
+
```

Comment:
Out of curiosity: Why are there *.d.ts files? Can't they be generated using `tsc`?

Reply:
They are necessary for the npm package, as far as i know they cannot be generated

Reply:
The typescript compiler can generate those files with [`"declaration": true`](https://www.typescriptlang.org/tsconfig#declaration) inside the `tsconfig.json`. And setting the [main](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#main) and [types](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package) inside `package.json` should instruct consumers of the package to look there for their imports.

Maybe this shouldn't be part of this PR but generating those files would reduce the complexity of the package nevertheless, maybe I'll look into it :wink:

Reply:
it works, very nice. Thx!

Reply:
I think it would be easier to put all generated files into the `dist` directory and set those as `main` and `types`.
Currently the d.ts is generated into the root of the repository/package.